doc(AIP-4114): Add documentation for ECP and new env var default. (#1451)
googleapis/aip: aip/auth/4114.md

  ...
  -----END PRIVATE KEY-----

### Obtaining the Default Device Certificate via ECP

The Enterprise Certificate Proxy (ECP) project is the newly recommended way to
procure device certificates. It has two major advantages compared to the legacy
SecureConnect mechanism:

1.  Allows usage of enterprise certs and private keys stored in native keystores and TPMs
    instead of relying on self-signed certs.
1.  Delegates signing operations to keystores, so private keys never leave the security realm.

Please see [ECP Public Documentation][2] for details on ECP configuration. ECP is

> fully supported in Google Golang libraries and gCloud SDK as of 2023.

Reviewer: Are we sure we want to mention which languages/surfaces support it here and not in say the doc site where there are already examples and better means of expressing that? Otherwise we will need to keep this updated whenever a new language/surface supports it.

That said, I'm not opposed to including it if no where else is reasonable - it's good to know and not have to ask around.

Author: This is a good point - I will omit the last part since the AIP is a recommendation and not a status tracker. Ideally there should be a separate public documentation today on which feature is supported in various clients by language, but no such thing exists AFAIK.

Reviewer: Yep I agree and TBH the client library AIPs don't have this level of tracking though we regularly ask each other which language supports which feature.

**GOOGLE_API_USE_CLIENT_CERTIFICATE**: If **"true"**, device certificate
authentication will be supported as described in the general guidance. If
**"false"**, the device certificate **must** not be used, even if specified by

> the user. The default value **should** be "true" as of May 3, 2024. Users who

Reviewer: ```suggestion
the user. The default value is "true" as of May 3, 2024. Users who
```

Is it in fact true everywhere? If so, just make it a statement. By using a requirement word like "should" we leave it open to ambiguity i.e. we think it should be but maybe it isn't. Notice the original text was matter-of-fact: "the default value will be 'false'"

Author: The wording "should" is intentional here, as not all supported languages have migrated to true by default (only Golang client has made the switch so far). Java and Python still needs to flip the switch, but we have no ETA on that as of now.
